sceptre/template.py

```diff
@@ -110,11 +110,16 @@ def _call_sceptre_handler(self):
 
         try:
             body = module.sceptre_handler(self.sceptre_user_data)
-        except AttributeError:
-            raise TemplateSceptreHandlerError(
-                "The template does not have the required "
-                "'sceptre_handler(sceptre_user_data)' function."
-            )
+        except AttributeError as e:
+            if 'sceptre_handler' in e.message:
+                raise TemplateSceptreHandlerError(
+                    "The template does not have the required "
+                    "'sceptre_handler(sceptre_user_data)' function."
+                )
+            else:
+                raise TemplateSceptreHandlerError(
```

Comment:
Could we switch this line to just `raise`? Python will raise the original exception, which I think is the preferred behaviour here. The template is written by the user so I think it makes sense to bubble up the same error that they would see if they compiled the template without Sceptre.

There's an example of an re-raising an exception [here](https://docs.python.org/3/tutorial/errors.html#raising-exceptions).